google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -408,17 +408,18 @@ private long getMaxBatchBytes() {
    * should be invoked prior to deleting the {@link Publisher} object in order to ensure that no
    * pending messages are lost.
    */
-  public void shutdown() throws Exception {
+  public void shutdown() {
     if (shutdown.getAndSet(true)) {
       throw new IllegalStateException("Cannot shut down a publisher already shut-down.");
     }
     if (currentAlarmFuture != null && activeAlarm.getAndSet(false)) {
       currentAlarmFuture.cancel(false);
     }
     publishAllOutstanding();
-    messagesWaiter.waitNoMessages();
     for (AutoCloseable closeable : closeables) {
-      closeable.close();
+      ExecutorAsBackgroundResource executorAsBackgroundResource =
```

Comment:
`closable.close()` calls `shutdown()` under the covers. There's no need to cast to a `ExecutorAsBackgroundResource`

Reply:
@sduskis changes done